Tests/DeepLearningTests/DatasetTests.swift

```diff
@@ -0,0 +1,154 @@
+        // host data types, which is not handled correctly in tracing.
+        let evens: Dataset = dataset.filter { Tensor($0 % 2) .== Tensor(0) }
+        XCTAssertEqual([0, 2, 4], evens.flatMap { $0.scalars })
+    }
+
+    func testParallelMap() {
+        let scalars = Tensor<Float>(rangeFrom: 0, to: 5, stride: 1)
+        let dataset = Dataset(elements: scalars)
+        let addedOne: Dataset = dataset.map(parallelCallCount: 5) { $0 + 1 }
+        XCTAssertEqual([1, 2, 3, 4, 5], addedOne.flatMap { $0.scalars })
+        // Use '.==' in the following closure to avoid any conversions to
+        // host data types, which is not handled correctly in tracing.
+        let evens: Dataset = dataset.filter { Tensor($0 % 2) .== Tensor(0) }
+        XCTAssertEqual([0, 2, 4], evens.flatMap { $0.scalars })
+    }
+
+    func testMapToDifferentType() {
+        let scalars = Tensor<Float>(rangeFrom: 0, to: 5, stride: 1)
+        let dataset = Dataset(elements: scalars)
+        let shuffled = dataset.shuffled(sampleCount: 5, randomSeed: 42)
+        XCTAssertEqual([0, 4, 1, 3, 2], shuffled.map { $0.scalar! })
+        let evens = shuffled.map { Tensor($0 % 2) .== Tensor(0) }
+        XCTAssertEqual([true, true, false, false, true], evens.map { $0.scalar! })
+    }
+
+    func testSingleValueBatched() {
+        let scalars = Tensor<Float>(rangeFrom: 0, to: 5, stride: 1)
+        let dataset = Dataset(elements: scalars)
+        let batched = dataset.batched(2)
+
+        var iterator = batched.makeIterator()
+        XCTAssertEqual([0, 1], iterator.next()!.scalars)
+        XCTAssertEqual([2, 3], iterator.next()!.scalars)
+        XCTAssertEqual([4], iterator.next()!.scalars)
+    }
+
+    func testDoubleValueDatasetIteration() {
+        let scalars1 = Tensor<Float>(rangeFrom: 0, to: 5, stride: 1)
+        let scalars2 = Tensor<Int32>(rangeFrom: 5, to: 10, stride: 1)
+        let datasetLeft = Dataset(elements: scalars1)
+        let datasetRight = Dataset(elements: scalars2)
+        var i: Int = 0
+        for pair in zip(datasetLeft, datasetRight) {
+            XCTAssertEqual(scalars1[i].array, pair.first.array)
+            XCTAssertEqual(scalars2[i].array, pair.second.array)
+            i += 1
+        }
+    }
+
+    static var allTests = [
+        ("testMultiValue", testMultiValue),
+        ("testSingleValueManualIterator", testSingleValueManualIterator),
+        ("testDatasetIteration", testDatasetIteration),
+        ("testSingleValueTransformations", testSingleValueTransformations),
+        ("testSingleValueHOFs", testSingleValueHOFs),
+        ("testParallelMap", testParallelMap),
+        ("testMapToDifferentType", testMapToDifferentType),
+        ("testSingleValueBatched", testSingleValueBatched),
+        // Currently broken even in TensorFlow ...
+        // This will be easier to fix once everything is moved ...
+        // ("testDoubleValueDatasetIteration", testDoubleValueDatasetIteration),
```

Comment:
I think commenting out the test here won't prevent the test from being from on macOS, so I'd recommend commenting out `func testDoubleValueDatasetIteration`. IIRC, XCTest on macOS doesn't use the `allTests` mechanism.